### DISCUSSION TITLE
What has changed in block structure ?

### DISCUSSION THREAD

**Discussion by sczembor** [casper-network/condor-info#20] [Q&A] [answered]
What is the primary breaking change of block structure?

**Comment by sczembor** [comment, marked as answer]
### Block structure:

The block structure will change with the condor release, however the historical blocks will still keep the old structure so it is important for all the parsers to support both the formats. 

**Pre-Condor (1.5.6):**

```
block
├── hash
├── header
│   ├── parent_hash
│   ├── state_root_hash
│   ├── body_hash
│   ├── random_bit
│   ├── accumulated_seed
│   ├── era_end 
│   ├── timestamp
│   ├── era_id
│   ├── height
│   └── protocol_version
├── body
└── proofs
```

**Post-Condor (2.0.0):**

```
block_with_signatures
├── block (Version2)
│   ├── hash
│   ├── header
│   │    ├── parent_hash
│   │    ├── state_root_hash
│   │    ├── body_hash
│   │    ├── random_bit
│   │    ├── accumulated_seed
│   │    ├── era_end 
│   │    ├── timestamp
│   │    ├── era_id
│   │    ├── height
│   │    ├── protocol_version
│   │    ├── current_gas_price <-- NEW
│   │    ├── last_switch_block_hash <-- NEW
│   │    └── proposer <-- Moved from body
│   └── body
└── proofs
```

**Key Changes:**

* **Versioning:** 
  * Introduction of explicit block structure versioning (`Version2`).
  * Pre Condor blocks are marked as `Version1`.

* **Header:**
  * New field: `current_gas_price`
  * New field: `last_switch_block_hash`
  * Moved Field: `proposer` moved from body.

* **Body:**
  * Now includes transactions and reward details rather than `deploy_hash`es

* **Proofs:**
  * Moved to top level (under `block_with_signatures`).

**Comment by ghost** [comment]
[Link](https://github.com/casper-network/condor-info/blob/main/resources/rpc-schema/chain_get_block.json) to GetBlock JSON-RPC schema.
